app/Http/Controllers/TransactionsController.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Jobs\ImportProcessJob;
+use App\Models\ImportProcess;
+use App\Models\Transaction;
+use App\Models\User;
+use Illuminate\Contracts\Pagination\LengthAwarePaginator;
+use Exception;
+use Illuminate\Database\Eloquent\Collection;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Validator;
+use PhpOffice\PhpSpreadsheet\Reader\Csv;
+
+class TransactionsController extends Controller
+{
+    /**
+     * Gets a list of transactions.
+     * Ideally this would be limited to the user and with a date range
+     * @param  Request  $request
+     * @return array
+     */
+    public function list(Request $request): array
+    {
+        /** @var User $user */
+        $user = $request->user();
+        $totalBalance = $user->transactions()->sum('amount');
+        $data = $user->transactions()->orderBy('date', 'desc')->paginate(50);
+        return ['totalBalance' => $totalBalance, 'transactions' => $data];
+    }
+
+    public function create(Request $request)
+    {
+        /** @var User $user */
+        $user = $request->user();
+        //validation for inputs
+        $validator = Validator::make(
+            $request->all(),
+            [
+                'label' => 'required',
+                'amount' => 'required|numeric',
+                'date' => 'required|date'
+            ]
+        );
+        if ($validator->fails()) {
+            return $validator->errors();
+        }
+        $transaction = new Transaction();
+        $transaction->label = $request->label;
+        $transaction->amount = $request->amount;
+        $transaction->date = $request->date;
+        $user->transactions()->save($transaction);
+        return $transaction->id;
+    }
+
+    /**
+     * @param  Request  $request
+     * @param $id
+     * @return bool|string
+     */
+    public function update(Request $request, $id)
+    {
+        //validation for inputs
+        $validator = Validator::make(
+            $request->all(),
+            [
+                'label' => 'required',
+                'amount' => 'required|numeric',
+                'date' => 'required|date'
+            ]
+        );
+        if ($validator->fails()) {
+            return $validator->errors();
+        }
+        /** @var User $user */
+        $user = $request->user();
+        /** @var Transaction $transaction */
+        $transaction = $user->transactions()->findOrFail($id);
+        $transaction->label = $request->label;
+        $transaction->amount = $request->amount;
+        $transaction->date = $request->date;
+        return $transaction->save();
+    }
```

Comment:
A Form Request and Model Binding would be preferred here.

https://laravel.com/docs/8.x/routing#route-model-binding